packages/compiler/lib/decorators.ts

```diff
@@ -404,6 +404,42 @@ export function $withUpdateableProperties({ program }: DecoratorContext, target:
   });
 }
 
+// -- @withoutOmittedProperties decorator ----------------------
+
+export function $withoutOmittedProperties(
+  { program }: DecoratorContext,
+  target: Type,
+  omitProperties: Type
+) {
+  if (omitProperties.kind == "TemplateParameter") {
+    // Silently return because this is a templated type
+    return;
+  }
+
+  if (!validateDecoratorTarget(program, target, "@withoutOmittedProperties", "Model")) {
+    return;
+  }
+
+  if (!validateDecoratorParamType(program, target, omitProperties, ["String", "Tuple"])) {
```

Comment:
Wouldn't an union should work too, that would be inline with typescript

Reply:
I'm OK with either approach.  Should I support one of them or both?

Reply:
I'd say then maybe just union for now and we can see tuple later if there is a need instead of removing in case this is causing issues

Reply:
Done